apis/kubedb/v1alpha1/mongodb_helpers.go

```diff
@@ -450,6 +458,36 @@ func (m *MongoDB) setDefaultProbes(podTemplate *ofst.PodTemplateSpec, mgVersion
 	}
 }
 
+// setDefaultAffinity
+func (m *MongoDB) setDefaultAffinity(podTemplate *ofst.PodTemplateSpec, key, value string) *ofst.PodTemplateSpec {
+	if podTemplate == nil || podTemplate.Spec.Affinity != nil {
+		return podTemplate
+	}
+
+	podTemplate.Spec.Affinity = &core.Affinity{
+		PodAntiAffinity: &core.PodAntiAffinity{
+			PreferredDuringSchedulingIgnoredDuringExecution: []core.WeightedPodAffinityTerm{
+				{
+					Weight: 100,
+					PodAffinityTerm: core.PodAffinityTerm{
+						LabelSelector: &metav1.LabelSelector{
+							MatchExpressions: []metav1.LabelSelectorRequirement{
+								{
+									Key:      key,
+									Operator: metav1.LabelSelectorOpIn,
+									Values:   []string{value},
+								},
+							},
+						},
+						TopologyKey: "failure-domain.beta.kubernetes.io/zone",
```

Comment:
This is more complicated. This key changed from 1.17 .

https://kubernetes.io/docs/reference/kubernetes-api/labels-annotations-taints/